submarine-server/server-core/src/main/java/org/apache/submarine/server/s3/Client.java

```diff
@@ -43,19 +45,38 @@
 
 public class Client {
   public MinioClient minioClient;
+  public static Map<String,MinioClient> minioClientFactory = new HashMap<String, MinioClient>();
+  public static Map<String,Client> clientFactory = new HashMap<String, Client>();
+
+  public static Client getClient(String endpoint) {
```

Comment:
Maybe you can use a `enum` class to replace inner factory class to avoid client double checks.

Reply:
Hi @cdmikechen , I agree with you.
But the enum type is more suitable for the predefined class (no argument to init). In our case, we need to init different users in testing, which cannot be predefined; that's why I use double-check locking.

Reply:
Java enum class can be used inside a class to create a constructor with parameters and as a singleton. Variables are only one of its basic uses.